src/domain/position/binProcessor.js

```diff
@@ -105,8 +106,15 @@ const processBins = async (bins, trx) => {
       array2.every((element) => array1.includes(element))
     // If non-prepare/non-commit action found, log error
     // We need to remove this once we implement all the actions
-    if (!isSubset([Enum.Events.Event.Action.PREPARE, Enum.Events.Event.Action.FX_PREPARE, Enum.Events.Event.Action.COMMIT, Enum.Events.Event.Action.RESERVE, Enum.Events.Event.Action.FX_RESERVE], actions)) {
-      Logger.isErrorEnabled && Logger.error('Only prepare/fx-prepare/commit actions are allowed in a batch')
+    if (!isSubset([
+      Enum.Events.Event.Action.PREPARE,
+      Enum.Events.Event.Action.FX_PREPARE,
+      Enum.Events.Event.Action.COMMIT,
+      Enum.Events.Event.Action.RESERVE,
+      Enum.Events.Event.Action.FX_RESERVE,
+      Enum.Events.Event.Action.TIMEOUT_RESERVED
```

Comment:
Don't we need separate actions like TIMEOUT_RESERVED and TIMEOUT_FX_RESERVED?

Reply:
Fx timeout reserved is in a follow up PR. This PR was scoped down to "port timeout functionality to batch".